19652 Fix mailing/physical address combo problems

NateJohnson-BAH opened the pull request:
## Description
The comparison tool is displaying a combination of the physical address and mailing address on the search results and profile pages. (now, we've also added the 'school locations' section of the profile page to this PR)
https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/vets.gov-team/19652

## Testing done
Confirmed locally in UI, via React Devtools to verify state and props were using correct info across various institutions

## Screenshots
Search Results showing in country, and out of country cards with appropriate info:
<img width="749" alt="Screen Shot 2019-09-27 at 12 52 13 PM" src="https://user-images.githubusercontent.com/52166695/65787060-16db7a00-e126-11e9-8a34-aef4b56e7e5b.png">


Example institution profile page showing non-US country, as expected
<img width="1007" alt="Screen Shot 2019-09-27 at 12 52 28 PM" src="https://user-images.githubusercontent.com/52166695/65787074-235fd280-e126-11e9-9114-7e35d36bbcb3.png">

School locations table for non-US institution:
<img width="1021" alt="Screen Shot 2019-09-27 at 12 52 43 PM" src="https://user-images.githubusercontent.com/52166695/65787102-37a3cf80-e126-11e9-8d40-a60e1545cfa4.png">

School locations table for US institution with zip code:
<img width="1022" alt="Screen Shot 2019-09-27 at 12 52 58 PM" src="https://user-images.githubusercontent.com/52166695/65787137-44282800-e126-11e9-95b3-f014fd2d117d.png">

## Acceptance criteria
- [x] Desired behavior:
1. If the institution is in the US or US Territory
a. The physical address city & state are displayed on the search results card.
  i. if only city, or only state is provided, show only relevant formatted information
  ii. if no physical address information is available, show a blank string
b. The physical address city & state are displayed on the profile page.
  i. if only city, or only state is provided, show only relevant formatted information
  ii. if no physical address information is available, the icon with location info is not displayed
c. The physical address city & state & zipcode are displayed on the profile page 'school locations' table.
  i. if only city, or only state is provided, show only relevant formatted information
  ii. if no physical address information is available, show a blank string
 iii. show physical zip code, if available, but after city, state (if available)
2. If the institution is foreign
a. The physical address city & country are displayed on the search results card.
  i. if only city, or only country is provided, show only relevant formatted information
  ii. if no physical address information is available, show a blank string
b. The physical address city & country are displayed on the profile page.
  i. if only city, or only country is provided, show only relevant formatted information
  ii. if no physical address information is available, the icon with location info is not displayed
c. The physical address city & country are displayed on the profile page 'school locations' table.
  i. if only city, or only country is provided, show only relevant formatted information
  ii. if no physical address information is available, show a blank string
 
## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

NateJohnson-BAH committed:
profile and search results now ony use physical City
 to avoid mixing with mailing addresses

NateJohnson-BAH committed:
Add conditional logic to profile to display state or
country depending on if country is USA

zurbergram commented on src/applications/gi/components/profile/HeadingSummary.jsx:
```diff
@@ -81,9 +81,18 @@ class HeadingSummary extends React.Component {
-              <IconWithInfo icon="map-marker" present={it.city && it.country}>
-                {it.city}, {it.state || it.country}
-              </IconWithInfo>
+              {it.country === 'USA' ? (
```
This variable name is not great

NateJohnson-BAH replied:
Agreed. If I made more changes to the file, I would have fixed it. Didn't want to cause more potential issues. :)

zurbergram reviewed (approved):
LGTM

shawkey-daniel commented:
The first example doesn't look right. Is there a Panama, FL?

NateJohnson-BAH commented:
@shawkey-daniel Yes, that's what is in the data:

![Screen Shot 2019-09-23 at 11 39 10 AM](https://user-images.githubusercontent.com/52166695/65440518-d793ed00-ddf6-11e9-807b-f7700739e551.png)

shawkey-daniel commented:
> @shawkey-daniel Yes, that's what is in the data:
> 
> ![Screen Shot 2019-09-23 at 11 39 10 AM](https://user-images.githubusercontent.com/52166695/65440518-d793ed00-ddf6-11e9-807b-f7700739e551.png)

I think that's bad data, but that's not part of this defect. Thanks.

NateJohnson-BAH commented:
In an attempt to simplify and force correct data usage, we have defaulted to 'physicalState' and physicalCity' for both search and profile pages on a given institution. Also, I added conditional logic to only show data that is available. For example, if we only have a city, that is all that is displayed. If we have a city, state then we display such. If the country is not USA, we display city and country. If the data seems incorrect, it is most likely bad data in the source weams file. 

Updated search results with more varied data to display changes:
<img width="790" alt="Screen Shot 2019-09-23 at 12 22 01 PM" src="https://user-images.githubusercontent.com/52166695/65443975-1331b580-ddfd-11e9-89fc-2d83d02e021e.png">

http://localhost:3001/gi-bill-comparison-tool/profile/11816110 is the example link given in the defect. This is what it looks like now. 'PANAMA' is listed as the physicalCity in the weams data, so is correct based on current logic. 
![Screen Shot 2019-09-23 at 12 26 52 PM](https://user-images.githubusercontent.com/52166695/65444141-73285c00-ddfd-11e9-9420-ecfd253a785f.png)

NateJohnson-BAH committed:
added physicalState to always use correct data

NateJohnson-BAH committed:
created function for conditional rendering of location info in profile header

shawkey-daniel reviewed (changes requested):
Can we get normalizedAttributes into a single location instead of repeating the code in two files?

NateJohnson-BAH committed:
remove one copy of 'normalizedAttributes' reducer function

shawkey-daniel reviewed (approved):
LGTM

NateJohnson-BAH committed:
changed let to const in reducer

NateJohnson-BAH committed:
Merge branch 'master' of github.com:department-of-veterans-affairs/vets-website into 19652-addressProblems

NateJohnson-BAH committed:
Merge branch 'master' of github.com:department-of-veterans-affairs/vets-website into 19652-addressProblems

NateJohnson-BAH committed:
Moved 'normalizeAttributes' function to it's own file, to please tests and reduce duplication

NateJohnson-BAH committed:
added unit test for utility reducer function

NateJohnson-BAH committed:
Merge branch 'master' into 19652-addressProblems

NateJohnson-BAH committed:
Merge branch 'master' of github.com:department-of-veterans-affairs/vets-website into 19652-addressProblems

NateJohnson-BAH committed:
create helper function for locationInfo

NateJohnson-BAH committed:
Merge branch 'master' of github.com:department-of-veterans-affairs/vets-website into 19652-addressProblems

shawkey-daniel reviewed (changes requested):
Can we combine the two locationInfo?

NateJohnson-BAH committed:
added country and zip to assure physical attributes are used

NateJohnson-BAH committed:
created locationInfo helper function for school locations section of profile page

NateJohnson-BAH committed:
added zip and country to logic for school location table

NateJohnson-BAH committed:
moved locationInfo function to helper file, used variables for assigning location label

NateJohnson-BAH committed:
consolidated the scool location table function

shawkey-daniel reviewed (approved):
LGTM

NateJohnson-BAH committed:
updated logic for VetTec location info to use the same logic and format as non-VetTec profiles

zurbergram committed:
Merge branch 'master' of github.com:department-of-veterans-affairs/vets-website into 19652-addressProblems

zurbergram committed:
new dev changes

- move schoolLocationTableInfo to only file that uses it
- change function in utility.js to be an exported lambda
- cleaned up locationInfo logic

rianfowler commented on .gitignore:
```diff
@@ -32,3 +32,4 @@ temp
 .swo
 all-links.csv
 .tern-port
+*.iml
```
must be our first intelliJ user 👍

zurbergram committed:
Merge branch 'master' of github.com:department-of-veterans-affairs/vets-website into 19652-addressProblems

zurbergram committed:
missed updating import statement
